clippy_dev/src/update_lints.rs

```diff
@@ -191,11 +189,14 @@ pub fn rename(old_name: &str, new_name: &str, uplift: bool) {
         .into_iter()
         .map(Result::unwrap)
         .filter(|f| {
-            let name = f.path().file_name();
-            let ext = f.path().extension();
-            (ext == Some(OsStr::new("rs")) || ext == Some(OsStr::new("fixed")))
-                && name != Some(OsStr::new("rename.rs"))
-                && name != Some(OsStr::new("deprecated_lints.rs"))
+            let path = f.path();
+            path.extension().is_some_and(|ext| ext == "rs" || ext == "fixed")
+                && !(path.file_name().is_some_and(|n| n == "lib.rs")
+                    && path
+                        .parent()
+                        .and_then(|p| p.parent())
+                        .and_then(|p| p.file_name())
+                        .is_some_and(|name| name == "clippy_deprecated_lints"))
```

Comment:
This condition was already pretty complicated and now got even more complicated. Please split it up into local variables.